assets/src/edit-story/app/history/historyProvider.js

```diff
@@ -35,8 +37,15 @@ function HistoryProvider({ children, size }) {
     historyLength,
     undo,
     redo,
+    versionNumber,
   } = useHistoryReducer(size);
 
+  const setPreventUnload = usePreventWindowUnload();
+
+  useEffect(() => {
+    setPreventUnload('history', versionNumber - 1 > 0);
```

Comment:
One more important nuance here possible. You also need an additional unmount effect:

```
useEffect(() => {
  return () => setPreventUnload('history', false);
}, [setPreventUpload])
```

This would be the most precise way to unblock unloading even if the component was unmounted.